presto-geospatial/src/main/java/com/facebook/presto/geospatial/BingTile.java

```diff
@@ -34,6 +36,35 @@ private BingTile(int x, int y, int zoomLevel)
         this.zoomLevel = zoomLevel;
     }
 
+    @Override
+    public boolean equals(Object other)
+    {
+        if (this == other) {
+            return true;
+        }
+
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+
+        BingTile otherTile = (BingTile) other;
+        return this.x == otherTile.x
+            && this.y == otherTile.y
+            && this.zoomLevel == otherTile.zoomLevel;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(x, y, zoomLevel);
+    }
+
+    @Override
+    public String toString()
+    {
+        return "Bing tile [" + x + ", " + y + "] at level " + zoomLevel;
```

Comment:
nit: you can also use `com.google.common.base.MoreObjects.toStringHelper` as we use in other classes.

Reply:
Is this the value that gets returned from the query for the SQL object?

Reply:
@electrum , I don't know. How can I find out?